src/prefect/server/api/workers.py

```diff
@@ -552,19 +622,28 @@ async def read_workers(
     offset: int = Body(0, ge=0),
     worker_lookups: WorkerLookups = Depends(WorkerLookups),
     db: PrefectDBInterface = Depends(provide_database_interface),
-) -> List[schemas.core.Worker]:
+) -> List[schemas.responses.WorkerResponse]:
     """
     Read all worker processes
     """
     async with db.session_context() as session:
         work_pool_id = await worker_lookups._get_work_pool_id_from_name(
             session=session, work_pool_name=work_pool_name
         )
-        return await models.workers.read_workers(
+        orm_workers = await models.workers.read_workers(
             session=session,
             work_pool_id=work_pool_id,
             worker_filter=workers,
             limit=limit,
             offset=offset,
             db=db,
         )
+        workers_response = []
+        for orm_worker in orm_workers:
+            worker_response = schemas.responses.WorkerResponse.from_orm(orm_worker)
```

Comment:
Is it worth overriding the `WorkerResponse.from_orm` and `WorkPoolResponse.from_orm` methods to attach the `status` attr? Seems like something those classes should own rather than it being one-off in the route itself.

Reply:
Yeah, good call. I've moved ownership of status determination to their respective classes.